source/codegen/templates/service_helpers.mako

```diff
@@ -19,7 +20,11 @@ ${initialize_input_params(function_name, parameters)}
       };
       uint32_t session_id = 0;
       const std::string& session_name = request->session_name();
+      %if 'custom_close' in function_data:
```

Comment:
Akshata has made some changes in her PR to extract close function to a mako variable. Please copy that here as well.

Also add a note-comment in the pull request here that this change is copied from other PR in features\ni-dmm branch and while merging to main we will resolve the conflict, so that they know that only one of these changes will go in main branch when we merge.

Reply:
I have copied the latest changes over here also. Thanks for putting it to notice.
I have left a comment on conversation and PR description both regarding conflict.